test/system/assignments_test.rb

```diff
@@ -22,7 +22,7 @@ class AssignmentsTest < ApplicationSystemTestCase
     assert page.has_text? 'Assignment was successfully created.'
     within '.table-striped' do
       assert page.has_text? 'Suggested'
-      assert page.has_link? 'superadmin@example.com'
+      assert page.has_link? 'superadmin@example.com', @user.email
```

Comment:
Why not either  `assert page.has_link? @user.email` or `assert page.has_link? 'superadmin@example.com'`?
http://www.rubydoc.info/github/jnicklas/capybara/Capybara/Node/Matchers#has_link%3F-instance_method

Reply:
This test should assure that the mail of the "creator" is shown at created_by.
With the first implementation we didn't noticed the "wrong" implementation, because we only checked if there is any mail (the superadmin@example.com was assigned to the client, and the test was green, because obviously the mail was displayed) and a client doesn't have the policy to create an assignment anyway. so two-in-one test :)

Reply:
If you run this test alone, you'll notice that you get a warning: 
`Unused parameters passed to Capybara::Queries::SelectorQuery : ["superadmin@example.com"]`
You cannot check two in one with `.has_link?` method.
The test checks if the email of the creator of the assignment is displayed. So `@user.email` should be sufficient. In the index view we don't display the email of the client. So the test is not mislead. In such a case, I think there would be an error, saying that there are two such links.